### ISSUE TITLE
give supervisors a site id

### ISSUE THREAD

**Issue by emiltin** [rsmp-nordic/rsmp_core#41]
Currently, only sites have a (site) id. I suggest giving all rsmp nodes an id, both sites and supervisors.

When a site connects to a supervisor, the supervisor can verify the site id to know who connecting. But the site has no way to know what supervisor it's connecting to, since the supervisor has no id and does not return any identifying info during the handshake. Currently the supervisor must return the id it received from the site, but the site already has this info.

When a site is connected to multiple supervisor, it's impractical that supervisors do not have a site id. For example, in log files, you have to resort to showing the ip.

**Comment by marcgarba** [comment]
Hello,

It was a question also we have on the "siteId" signification and use.
So I continue directly here the exchange on that point.

It is not really clear that a supervisor could not be also considered as a different site for this information.
In RSMP core, for the initial version message, it is written that the "siteId" must match, but we could consider it match what is expected for each side:
- the equipment send it's site name => the supervision test if it is the equipment expected.
- the supervision send it's scada name => the equipment test if it is the scada name corresponding to the network configuration supervisions defined for the client connection.

Currently in the RSMP simulators, the content of the "siteId" information is not verified, and we could have differents sites names on each side without being rejected.

Else, what is the meaning to have multiples sites name for one physical equipment ?

Thanks.

**Comment by otterdahl** [comment]
The reason for having both siteId as well as componentId is because of flexibility and supporting various types of equipment and network infrastructures.

* A single RSMP connection can be shared between several sites.
* A site can contain several physical equipments

For TLC's, we have just a single RSMP connection for each site, and each site only includes a single TLC, for simplicity.

But we have cases for other types of equipment where many physical sites shares a single RSMP connection. In that case, RSMP is used between systems, and not directly to the site.

**Comment by marcgarba** [comment]
Thanks for your reply and explanations.

But about the initial point explained by Emilitin in its first post :

> When a site connects to a supervisor, the supervisor can verify the site id to know who connecting. But the site has no way to know what supervisor it's connecting to, since the supervisor has no id and does not return any identifying info during the handshake. Currently the supervisor must return the id it received from the site, but the site already has this info.

it seems interesting to give a siteId for a supervisor and allow it,
so that the equipment can know with who it is connected (and eventually check it). And as already mentioned, interesting to analyse logs.

**Comment by emiltin** [comment]
With RSMP 4 all nodes have an id.
I'm also wondering if end-to-end encryption will require the supervisor to identify with an id?

**Comment by marcgarba** [comment]
There are many cases where to give a name to supervisor is interesting. And for EndToEndEncryption it is even required...

**Comment by emiltin** [comment]
I suggest that we define in core 3.3.0 that supervisor must have an id. The Version response send by the supervisor must includes "supervisorId" instead of "siteId", and is set to the id of the supervisor:. I don't think it needs to be an array:

```
     "supervisorId": "KK+SU001"
```

The initial Version request from the site must have a very stable format, so we cannot remove or change fields, but the response can be modified, I think, since the core version has been decided when it's sent.

Btw, note that in core 3.3.0 a TCP connection can only be used for a singe site. So the siteId array will contain just one item.

**Comment by marcgarba** [comment]
To preserve the current meaning of siteId I personnaly discovered (when received by the equipment to verify that the supervision is addressing to the correct equipment), a ‘new’ dedicated field could be added.

However, it would also be necessary to cover the case of leader<->follower inter-equipment communication, with a 'new' attribute name that is identical. The term ‘supervision’ seems too restrictive. A term that can be used for both would be needed.

As a supervisor is also a "leader" of the communication.... just "leaderId" ?

**Comment by emiltin** [comment]
> However, it would also be necessary to cover the case of leader<->follower inter-equipment communication, with a 'new' attribute name that is identical. The term ‘supervision’ seems too restrictive. A term that can be used for both would be needed.

Good point. RSMP 4 uses the term 'node' to cover both site and supervisor. We could use that, but then it would make more sense to use it for both the request and the response. So we could:

- keep using siteId in both directions, and just require that it always contain the sender's id
- add a new nodeId, and leave siteId as it is

**Comment by marcgarba** [comment]
or "DriverId" ?
covering supervision & leader equipment...

Of course, a single attribute "siteId" always corresponding to the message sender would be simpler.
But what about the change from the existing meaning?
Or, since we know we are in core 3.3, we could treat it differently and add a new "recipientId" attribute to confirm correct routing with supervision...

A very small point, but one worth thinking about nonetheless! ;-)
